cpp/src/string/nvcategory_util.cu

```diff
@@ -0,0 +1,170 @@
+
+#include "nvcategory_util.cuh"
+
+
+#include <NVCategory.h>
+#include <NVStrings.h>
+#include "rmm/rmm.h"
+
+#include "utilities/error_utils.hpp"
+#include "utilities/nvtx/nvtx_utils.h"
+
+
+gdf_error create_nvcategory_from_indices(gdf_column * column, NVCategory * nv_category){
+
+	if(nv_category == nullptr ||
+			column == nullptr){
+		return GDF_INVALID_API_CALL;
+	}
+
+	if(column->dtype != GDF_STRING_CATEGORY){
+		return GDF_UNSUPPORTED_DTYPE;
+	}
+
+	NVStrings * strings = nv_category->gather_strings(static_cast<nv_category_index_type *>(column->data),
+			column->size,
+			DEVICE_ALLOCATED);
+
+	NVCategory * new_category = NVCategory::create_from_strings(*strings);
+	new_category->get_values(static_cast<nv_category_index_type *>(column->data),
+			DEVICE_ALLOCATED);
+
+	//This is questionable behavior and should be reviewed by peers
```

Comment:
In general cuDF needs a memory ownership / lifetime redesign.  For now, I think you should add test coverage of risky usage rather than adding disconcerting comments.